cachix/src/Cachix/Client/Servant.hs

```diff
@@ -24,3 +26,7 @@ isErr (FailureResponse resp) status
 #endif
   | responseStatusCode resp == status = True
 isErr _ _ = False
+
+-- | Convert 'NoContent' to '()' in order to silence a needless warning.
+discardNoContent :: Functor f => f NoContent -> f ()
+discardNoContent = void
```

Comment:
I like to avoid plain `void` because in `void m`, if `m :: f NoContent` it's harmless but if changed to `m :: f Either ErrInfo NoContent`, you suddenly lose important information. The compiler can help, but you need to declare that you don't want to destroy information, which can be done in this way.

Alternatively, we can require type applications like `void @_ @NoContent $ m`, use `ScopedTypeVariables` and `(_ :: NoContent) <- m`, or perhaps even `NoContent <- m` although that's needlessly strict.

Reply:
I've picked `ScopedTypeVariables` for now.